components/PoolCard/index.js

```diff
@@ -24,76 +24,80 @@ import {
   LogoContainer,
   PoolCardFooter,
   PoolCardFooterButton,
-  PoolRewardData,
+  PoolCardInfoContainer,
   PoolSubtitle,
   PoolTitle,
   Wrapper,
-} from './styled'
+} from "./styled";
 
 export default function PoolCard({
   stakingPoolData,
+  vaultOption,
   setIsStakeAction,
   setShowApprovalModal,
   setShowClaimModal,
   setShowActionModal,
 }) {
-  const { transactions } = useTransactions()
-  const { account, initAccount } = useAppContext()
-  const { tokenAllowance } = useTokenAllowance()
-  const { tvlInUsd } = useAPRandVexPrice()
+  const { transactions } = useTransactions();
+  const { account, initAccount } = useAppContext();
+  const { tokenAllowance } = useTokenAllowance(vaultOption);
+  let disableClaimButton = true;
+  const color = colors.orange;
 
-  const color = colors.orange
+  const currentStakeInUsd = useMemo(() => {
+    if (
+      !stakingPoolData.poolData.tvlInUsd ||
+      stakingPoolData.poolData.tvlInUsd.eq(0) ||
+      !stakingPoolData.userData.currentStake
```

Comment:
```suggestion
         !stakingPoolData.poolData.tvlInUsd
      || stakingPoolData.poolData.tvlInUsd.eq(0)
      || !stakingPoolData.userData.currentStake
```

this is more readable

Reply:
@andersonaguiar due to the existing code style, ignore this suggestion but keep this style in mind for future repos. goal here is to just get the feature working and refactor later